### PR TITLE
expression: open CAST string as real push down switcher

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

### What problem does this PR solve? 

This PR opens the push-down switcher of the CAST string as real.

Additional, this PR fixes the https://github.com/pingcap/tidb/issues/14326.

### What is changed and how it works?

The https://github.com/tikv/tikv/pull/6390 has fix the inconsistent with TiDB.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: expression/builtin_arithmetic.go [hunk 1]
```diff
@@ -273,13 +273,16 @@ func (s *builtinArithmeticPlusRealSig) Clone() builtinFunc {
 }
 
 func (s *builtinArithmeticPlusRealSig) evalReal(row chunk.Row) (float64, bool, error) {
-	a, isNull, err := s.args[0].EvalReal(s.ctx, row)
-	if isNull || err != nil {
-		return 0, isNull, err
+	a, isLHSNull, err := s.args[0].EvalReal(s.ctx, row)
+	if err != nil {
```
Comment (XuHuaiyu) [inline expression/builtin_arithmetic.go]: Why change this?

Comment (lonng) [reply expression/builtin_arithmetic.go]: All of the expressions which have two arguments in TiDB have this pattern, they have inconsistent behavior with MySQL.
```
create table t (a varchar(10));
insert into t values ('2007');
select * from t where NULL + pow(7020, a);
```
The above statement should report an error.

**Comment by lonng** [comment]
Note: this PR should be merged with https://github.com/tikv/tikv/pull/6390 together.

**Pull request merged**